engine/src/main/java/org/terasology/rendering/dag/stateChanges/SetRenderBufferMask.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.dag.stateChanges;
+
+import org.terasology.assets.ResourceUrn;
+import org.terasology.rendering.opengl.BaseFBOsManager;
+import org.terasology.rendering.opengl.FBO;
+import org.terasology.rendering.opengl.FBOManagerSubscriber;
+import com.google.common.base.Objects;
+import org.terasology.rendering.dag.RenderPipelineTask;
+import org.terasology.rendering.dag.StateChange;
+
+/**
+ * This StateChange generates the tasks that set and reset render masks.
+ *
+ * Every FBO can either allow or forbid OpenGL from writing to its Color buffer, DepthStencil buffer and Light buffer.
+ * This StateChange modifies the FBO to mask or unmask these three buffers, independent of each other.
+ * Note that this StateChange assumes the given FBO has the required attachments, and trying to unmask the
+ * depth buffer or light buffer for a normal FBO with default configuration -might- lead to undefined behaviour.
+ */
+public final class SetRenderBufferMask implements FBOManagerSubscriber, StateChange {
+    private ResourceUrn fboName;
+    private BaseFBOsManager fboManager;
+
+    private boolean renderToColorBuffer;
+    private boolean renderToDepthBuffer;
+    private boolean renderToLightBuffer;
+
+    private SetRenderBufferMask defaultInstance;
+    private SetRenderBufferMaskTask task;
+
+    /**
+     * Creates an instance of this class with the given parameters.
+     *
+     * @param fboName A ResourceUrn identifying the FBO whose render masks have to be modified.
```

Comment:
I'd write:

`@param fboName A ResourceUrn identifying the FBO whose render masks have to be modified - usually only the writeOnlyGBuffer FBO. `

In practice right now this state change is used also for the read-only gBuffer, but that's an aberration that will be corrected eventually.